src/lib.rs

```diff
@@ -1,6 +1,560 @@
+                    prev_char = '\\';
+                },
+                //deletion
+                '<' => {
+                    if prev_char == '^'{
+                        my_opp_vector.reverse();
+                    }
+                    let insertionOpp : StringDiffOp = StringDiffOp{
+                        kind : StringDiffOpKind::Delete(top_string.chars().nth(top_str_len-1).unwrap()),
+                        index : (top_str_len as isize) - 1
+                    };
+                    top_str_len -= 1;
+                    my_opp_vector.push(insertionOpp);      
+                    prev_char = '<';
+                },
+                _ => { 
+                    panic!("UNRECOGINZED SYMBOL OPERATION !")
+                }
+            }
+
+        }
+        return my_opp_vector;
+        //println!("{:?}" , my_opp_vector)
+    }
+   
+}
+impl StringDiffAlgorithm for LevenshteinDistance {
+    //fn diff<'a>(s1: &'a str, s2: &'a str) -> Vec<StringDiffOp>{}
+    fn diff<'a>(&self, s1: &'a str, s2: &'a str) -> Vec<StringDiffOp> {
+        
+        let first_string_len : usize = s1.len();
+        let second_string_len : usize = s2.len();
+        
+        let mut dist_vector = vec![vec![0usize ; second_string_len + 1]; first_string_len + 1];
+        let mut dir_vector: Vec<Vec<char>> = vec![vec![' ' ; second_string_len + 1]; first_string_len + 1];
+
+        for i in 0..first_string_len+1{
+            dist_vector[i][0] = i;
+        }
+        for j in 0..second_string_len+1{
+            dist_vector[0][j] = j;
+        }
+
+        dir_vector[0][0] = '\\';
+        for j in 1..second_string_len+1{
+            dir_vector[0][j] = '<';
+        }
+        for i in 1..first_string_len+1{
+            dir_vector[i][0] = '^';
+        }
+
+        let mut sub_cost : usize = 0;
+        for i in 1..first_string_len+1{
+            for j in 1..second_string_len+1{
+                if s1.chars().nth(i-1).unwrap() ==  s2.chars().nth(j-1).unwrap(){
+                    sub_cost = 0;
+                }
+                else{
+                    sub_cost = 1;
+                }
+                (dist_vector[i][j], dir_vector[i][j])  = LevenshteinDistance::min_dist_with_dir(
+                    dist_vector[i-1][j] + 1, //deletion
+                    dist_vector[i][j-1] + 1, //insertion
+                    dist_vector[i-1][j-1] + sub_cost); //substitution
+            }
+        }
+
+        //LevenshteinDistance::print_vector(&dist_vector);
+        //LevenshteinDistance::print_vector(&dir_vector);
+        /*
+        for row in dist_vector.iter(){
+            println!("{:?}", row)
+        }
+        for row in dir_vector.iter(){
+            println!("{:?}", row)
+        }
+        */
+
+        //println!("Made it here");
+        let test = LevenshteinDistance::get_operations(&dir_vector, s1, s2);
+        println!("{:?}", test);
+        //return dist_vector[first_string_len][second_string_len];
+        return test;
```

Comment:
There's some debug stuff leftover that should be taken out, and then immediately return `LevenshteinDistance::get_operations(&dir_vector, s1, s2)` without creating a new variable